### ISSUE TITLE
Is it possible to build macOS Intel executable on macOS Apple Silicon machine?

### ISSUE THREAD

**Issue by pie6k** [ggml-org/whisper.cpp#1898]
I want to build an universal binary that will work both on Apple Silicon and Intel macOS.

I did experiment with Makefile:

```Makefile
ifndef UNAME_P
# UNAME_P := $(shell uname -p)
# force intel
UNAME_P := i368

endif

ifndef UNAME_M
# UNAME_M := $(shell uname -m)
# force intel
UNAME_M := x86_64

endif
```

By forcing `UNAME_P` and `UNAME_M` to match those on Intel machine, but the builded main file is stil ARM executable.

Thus I wonder if I need an actual Intel macOS machine to build it. 

Also I did check Releases section of this repo, but seems there are no macOS builds there.

**Comment by thebinarysearchtree** [comment]
I don't think so. I have both an intel mac and an m2 and compiled two binaries and then just switch between them depending on what machine they are running on.

**Comment by vade** [comment]
FWIW You can use `lipo` command line tool  to make fat binaries out of 2 separate architectures

**Comment by thebinarysearchtree** [comment]
Yeah, but you will still need an intel mac :)

**Comment by vade** [comment]
? you can cross compile on arm to x86_64 and vice versa

**Comment by thebinarysearchtree** [comment]
You are right, I am stupid.

**Comment by vade** [comment]
Nah friend, theres just a lot to keep track of and learn. You're fine !

**Comment by pie6k** [comment]
I think I figured it out:

Build for Apple Silicon:


```
make clean
make
```


Then rename main into main-arm64:

Build for Intel (x86_64):


```
make clean
arch -x86_64 make
```


Then rename main into main-x86_64:

Combine the two builds into a universal binary:


```
lipo -create -output whisper-universal main-arm64 main-x86_64
```

Ensure that the universal binary is working:


```
file whisper-universal
```
